server/controllers/stock/functions/rumer.function.js

```diff
@@ -0,0 +1,193 @@
+const _ = require('lodash');
+const moment = require('moment');
+
+const db = require('../../../lib/db');
+const core = require('../core');
+
+exports.getData = getData;
+
+// default report parameters
+const DEFAULT_PARAMS = {
+  csvKey : 'rumer',
+  filename : 'REPORT.RUMER.TITLE',
+  orientation : 'landscape',
+};
+
+/**
+ * @method getRumer
+ *
+ * @description
+ * This method builds the RUMER (Registre d’Utilisation des Médicaments Et Recettes) report by month
+ *
+ * GET /stock/rumer
+ */
+async function getData(reqQuery) {
+  const params = reqQuery;
+  params.depotUuid = params.depotUuid || params.depot_uuid;
+  params.exclude_out_stock = parseInt(params.exclude_out_stock, 10);
+  params.include_daily_balances = parseInt(params.include_daily_balances, 10);
+  params.condensed_report = parseInt(params.condensed_report, 10);
+
+  const data = {};
+  const headerReport = [];
+  const configurationData = [];
+
+  params.start_date = moment(new Date(params.start_date)).format('YYYY-MM-DD');
+  params.end_date = moment(new Date(params.end_date)).format('YYYY-MM-DD');
+
+  const startDate = parseInt(moment(params.start_date).format('DD'), 10);
+  const endDate = parseInt(moment(params.end_date).format('DD'), 10);
+
+  const lastDayPreviousMonth = moment(params.start_date).subtract(1, 'day').format('YYYY-MM-DD');
+
+  for (let i = startDate; i <= endDate; i++) {
+    headerReport.push(i);
+  }
+
+  data.params = params;
+  data.header = headerReport;
+
+  const parameterOpeningStock = {
+    depot_uuid : params.depotUuid,
+    dateTo : lastDayPreviousMonth,
+    includeEmptyLot : 1,
+  };
+
+  const parameterEndingStock = {
+    depot_uuid : params.depotUuid,
```

Comment:
`params.depot_uuid`

Reply:
The function is already used by the RUMER report module, so I do not want to change currently codes on the client or some existing modules